rcp/org.testeditor.rcp4.views.tcltestrun/src/org/testeditor/rcp4/views/tcltestrun/TclLauncherUi.xtend

```diff
@@ -0,0 +1,173 @@
+	@Inject TclGradleLauncher gradleLauncher
+
+	override boolean launch(IStructuredSelection selection, IProject project, String elementId, String mode,
+		boolean parameterize) {
+		val options = new HashMap<String, Object>
+		if (project.getFile("build.gradle").exists) {
+			return launchTest(selection, project, elementId, gradleLauncher, options)
+		}
+		if (project.getFile("pom.xml").exists) {
+			if (parameterize) {
+				val mavenOptions = uiCollectMavenOptions(project)
+				if (mavenOptions == null) {
+					return true // should an option always be selected?
+				}
+				options.putAll(mavenOptions)
+			}
+			return launchTest(selection, project, elementId, mavenLauncher, options)
+		}
+		logger.warn("gradle based launching test for tcl element='{}' failed, since file='build.gradle' was not found.",
+			elementId)
+		logger.warn("maven based launching test for tcl element='{}' failed, since file='pom.xml' was not found.",
+			elementId)
+		return false
+	}
+
+	private def Map<String, Object> uiCollectMavenOptions(IProject project) {
+		val mavenProfiles = project.mavenGetProfilesWithUiFeedback
+		val dialog = new ElementListSelectionDialog(PlatformUI.workbench.activeWorkbenchWindow.shell, new LabelProvider)
+		dialog.setElements(mavenProfiles)
+		dialog.setTitle("Which maven profile should be used?")
+		if (dialog.open == Window.OK) {
+			val selectedProfile = dialog.result
+			return #{TclMavenLauncher.PROFILE -> selectedProfile.get(0)}
+		} else {
+			return null // cancelled
+		}
+	}
+
+	private def Iterable<String> mavenGetProfilesWithUiFeedback(IProject project) {
+		val Iterable<String>[] container = newArrayOfSize(1)
+		progressRunner.run([ monitor |
+			monitor.beginTask("Collect maven profiles", IProgressMonitor.UNKNOWN)
+			container.set(0, mavenLauncher.getProfiles(project))
+			monitor.done
+		])
+		return container.head
+	}
+
+	private def boolean launchTest(IStructuredSelection selection, IProject project, String elementId,
+		TclLauncher launcher, Map<String, Object> options) {
+		logger.info("Trying to launch launcherClass='{}' test execution for elementId='{}' in project='{}'",
+			launcher.class.simpleName, elementId, project)
+		progressRunner.run([ monitor |
+			monitor.beginTask("Test execution: " + elementId, IProgressMonitor.UNKNOWN)
+			val result = launcher.launchTest(selection, project, elementId, monitor, options)
+			project.refreshLocal(IProject.DEPTH_INFINITE, monitor)
+			val expectedFile = (result.get(TclLauncher.EXPECTED_FILE) as File)
+			if (expectedFile == null) {
+				logger.error("resulting expectedFile must not be null")
+			} else {
+				safeUpdateJunitTestView(elementId, expectedFile)
+			}
+			monitor.done
+		])
+		return true
+	}
+
+	/** 
+	 * provide test result file (either the one created by the test run, or, if absent/on error 
+	 * a default error file) that is imported into junit (and thus displayed) 
+	 * */
+	private def void safeUpdateJunitTestView(String elementId, File expectedFile) {
```

Comment:
I'm not very happy with this coding either. The reason for this moving result files about is that I want to prevent old test results to accidently appear. Using guava made the code a bit shorter but not really clean. I'm open for suggestions here (I did not quite get what's ment above...)